The failing macOS checks here look unrelated to this change.

On the macOS jobs the MLX test suite itself passes:

```
Ran 773 tests in 90.930s

OK (skipped=5)
```

and Linux (CPU + all CUDA variants), lint, and docs are green. The macOS job then fails in the separate `mpi_test_distributed.py` step with a segfault inside OpenMPI/PRRTE at MPI init — before any MLX code runs:

```
[...] *** Process received signal ***
[...] Signal: Segmentation fault: 11 (11)
[...] libprrte.3.dylib   construct_range + 36
[...] libprrte.3.dylib   prte_hwloc_base_get_topo_signature + 632
[...] libprrte.3.dylib   rte_init + 1940
[...] libprrte.3.dylib   prte_init + 1044
93754 Segmentation fault: 11  mpirun --bind-to none -host localhost:8 -np 8 -x DYLD_LIBRARY_PATH=/opt/homebrew/lib/ python python/tests/mpi_test_distributed.py
##[error]Process completed with exit code 139
```

The crash is in the `hwloc` topology probe during `prte_init`, so it looks like an OpenMPI/runner-environment issue rather than anything in this quantized-matvec change — all three macOS runners (14/15/26) hit it identically in the same run, while the actual MLX tests (including the new `out_vec_size < 8` cases) pass.

I've reopened the PR to re-run CI; the new run is currently waiting on workflow approval. Happy to rebase if that's easier.

--- a/mlx/backend/metal/kernels/fp_quantized.h
+++ b/mlx/backend/metal/kernels/fp_quantized.h
@@ -440,7 +440,7 @@ METAL_FUNC void fp_qmv_impl(
         auto wl = (const device uint8_t*)(ws + row * in_vec_size_w);
         const device auto* sl = scales + row * in_vec_size_g;
 
-        uint8_t s = sl[0];
+        U s = dequantize_scale<U, group_size>(sl[0]);
         result[row] += qdot<U, values_per_thread, bits>(wl, x_thread, s);
       }
 

--- a/python/tests/test_quantized.py
+++ b/python/tests/test_quantized.py
@@ -435,7 +435,7 @@ class TestQuantized(mlx_tests.MLXTestCase):
         key = mx.random.key(0)
         k1, k2 = mx.random.split(key)
         tests = product(
-            [256, 512, 67],  # M
+            [256, 512, 67, 5, 7],  # M -- 5, 7 exercise out_vec_size < 8 (#3762)
             [64, 256],  # N
             [0, 1, 3, 8],  # B
         )

